src/Gaffer/Box.cpp

```diff
@@ -468,11 +465,6 @@ BoxPtr Box::create( Node *parent, const Set *childNodes )
 				}
 			}
 		}
-		// reparent the child under the Box. it's important that we do this after adding the intermediate
-		// input plugs, so that when they are serialised and reloaded, the inputs to the box are set before
-		// the inputs to the nodes inside the box - see GafferSceneTest.ShaderAssignmentTest.testAssignShaderFromOutsideBox
```

Comment:
Maybe just a comment in the commit message explaining why this isn't the case anymore? Not a big deal though...